lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/RetrieveSourcePath.pm

```diff
@@ -92,19 +102,24 @@ sub retrieve_source_md5 {
     my $self = shift;
     $self->debug_message('Create destination MD5 path...');
 
-    my $source_md5 = $self->source_md5;
-    if ( not $source_md5) {
+    if ( not $self->source_file->md5_path_size ) {
         $self->debug_message('Source MD5 is not available! It will not be saved.');
         return 1;
     }
 
+    my $source_md5 = $self->_load_source_md5;
+    if ( not $source_md5 ) {
+        $self->debug_message('Source MD5 could not be loaded from $md5_path and therefore not storing it.');
```

Comment:
It might be nice if this debug message matched the structure of the surrounding messages. e.g., `Source MD5 could not be loaded!  It will not be saved.`.

Either way, including the literal string `$md5_path` in the message probably does not add much value.